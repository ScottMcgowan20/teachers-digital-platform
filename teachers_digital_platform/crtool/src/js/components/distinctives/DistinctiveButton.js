import React from "react";

import C from "../../constants"; 

export default class DistinctiveButton extends React.Component {
  handleOnClick() {
    this.props.distinctiveClicked(this.props.distinctive);
  }

  render() {

    let statusClassName = "o-dimension-section-bar_selector ";
    if (this.props.inProgress === C.STATUS_COMPLETE) {
      statusClassName += "is-complete"
    }
    else if (this.props.inProgress === C.STATUS_IN_PROGRESS) {
      statusClassName += "is-active"
    }

    return (
      <li className="o-dimension-section-bar_dimension">
        <a href="#" className={statusClassName} onClick={this.handleOnClick.bind(this)}>
            <span className="o-dimension-section-bar_selector-title">
                <span className="a-icon a-icon__large"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 651.7 1200" className="cf-icon-svg"><path d="M30 161c-16.5 0-30 13.5-30 30v827.8c0 16.5 13.5 30 30 30h591.7c16.5 0 30-13.5 30-30V343.7L469 161H30zm389.6 60v134.8c0 19.9 16.3 36.2 36.2 36.2h135.9V988.8H60V221h359.6z"/><path d="M123.8 768.6h394.8v50H123.8zM123.8 644h394.8v50H123.8zM123.8 519.5h394.8v50H123.8z"/><circle cx="194" cy="382.3" r="60"/></svg></span>
                &nbsp;{this.props.title}
                <span className="o-dimension-section-bar_selector-status">&nbsp;{this.props.inProgress}</span>
            </span>
            <span className="o-dimension-section-bar_selector-text">Covers core knowledge and skills in national standards</span>
        </a>
        <ul className="o-dimension-section-bar_dimension-details">
            <li className="h6">6 criteria</li><li className="h6">30 minutes</li>
        </ul>
    </li>
    );
  }
}
