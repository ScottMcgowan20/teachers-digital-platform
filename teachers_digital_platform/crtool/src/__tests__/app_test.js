import React from 'react';
import ReactDOM from 'react-dom';
import DistinctiveButton from '../js/components/distinctives/DistinctiveButton';
import localStorage from 'mock-local-storage';
import renderer from 'react-test-renderer';
import jest from 'jest';

describe ('initial state', () => {

  it('renders without crashing', () => {
    const component = renderer.create(
      <DistinctiveButton />,
    );
  });  

  it('renders without crashing 2', () => {
    const component = renderer.create(
      '<DistinctiveButton distinctiveClicked={this.distinctiveClicked.bind(this)} {...distinctiveProps}/>',
    );
  }); 
})