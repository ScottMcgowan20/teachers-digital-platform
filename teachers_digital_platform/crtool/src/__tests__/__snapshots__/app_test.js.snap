// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="block block__flush-top"
  >
    <h1>
      Curriculum Review
    </h1>
    <p>
      Middle School Example Curriculum
    </p>
    <p>
      Start the review by selecting a dimension. You do not need to complete all dimensions in one sitting. Youâ€™ll be able to download a dimension report for each dimension as well as a summary report at the end.
    </p>
    <div
      className="DistinctivesBlock"
    >
      <button
        onClick={[Function]}
      >
        <div
          className="Distinctive"
        >
          <div
            className="DistinctiveBoxFormat"
          >
            <h4>
              Content
               
            </h4>
            <p>
              Covers core knowledge and skills in content standards
              <br />
              <br />
              6 criteria
              <br />
              Est. time 30 min
            </p>
          </div>
        </div>
      </button>
      <button
        onClick={[Function]}
      >
        <div
          className="Distinctive"
        >
          <div
            className="DistinctiveBoxFormat"
          >
            <h4>
              Utility
               
            </h4>
            <p>
              Supports effective teaching
              <br />
              <br />
              7 criteria
              <br />
              Est. time 30 min
            </p>
          </div>
        </div>
      </button>
      <button
        onClick={[Function]}
      >
        <div
          className="Distinctive"
        >
          <div
            className="DistinctiveBoxFormat"
          >
            <h4>
              Quality
               
            </h4>
            <p>
              Accurate and well presented
              <br />
              <br />
              5 criteria
              <br />
              Est. time 30 min
            </p>
          </div>
        </div>
      </button>
      <button
        onClick={[Function]}
      >
        <div
          className="Distinctive"
        >
          <div
            className="DistinctiveBoxFormat"
          >
            <h4>
              Efficacy
               
            </h4>
            <p>
              Improves financial knowledge, skills, or behaviors
              <br />
              <br />
              3 criteria
              <br />
              Est. time 30 min
            </p>
          </div>
        </div>
      </button>
    </div>
    <div>
      <div>
        <h1 />
        <div
          className="block"
        >
          <h2>
            Content for elementary school
          </h2>
          <p>
            Curriculum content refers to the knowledge and skills covered by the curriculum. The content dimension is based on our analysis of existing national and select state financial education standards. An indicator is a concrete, detailed specification for each criteria. A component is an operational rule to help reviewers determine whether the curriculum meets each indicator. Some are essential and some are beneficial.
          </p>
          <h3>
            Instructions
          </h3>
          <p>
            Read through the scope and sequence of the curriculum, skim the lesson plans, and answer the following questions.
          </p>
          <div>
            DEBUG Values:
            <p>
              Curriculum Title: 
               
            </p>
            <p>
              Publication Date: 
               
            </p>
            <p>
              Grade Range: 
               
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
